fix TestISequencer_LongRecovery

https://github.com/voedger/voedger/actions/runs/14372813004/job/40298919008

```
WARNING: DATA RACE
  Write at 0x00c00030a1a8 by goroutine 471:
    github.com/voedger/voedger/pkg/isequencer.(*sequencer).stopFlusher()
        /home/runner/work/voedger/voedger/pkg/isequencer/impl.go:90 +0x104
    github.com/voedger/voedger/pkg/isequencer.(*sequencer).actualizer()
        /home/runner/work/voedger/voedger/pkg/isequencer/impl.go:384 +0x149
    github.com/voedger/voedger/pkg/isequencer.(*sequencer).startActualizer.gowrap1()
        /home/runner/work/voedger/voedger/pkg/isequencer/impl.go:83 +0x4f
  
  Previous read at 0x00c00030a1a8 by goroutine 288:
    github.com/voedger/voedger/pkg/isequencer.(*sequencer).cleanup()
        /home/runner/work/voedger/voedger/pkg/isequencer/impl.go:539 +0x158
    github.com/voedger/voedger/pkg/isequencer.(*sequencer).cleanup-fm()
        <autogenerated>:1 +0x33
    github.com/voedger/voedger/pkg/isequencer_test.TestISequencer_LongRecovery()
        /home/runner/work/voedger/voedger/pkg/isequencer/isequencer_test.go:953 +0x652
    testing.tRunner()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:1792 +0x225
    testing.(*T).Run.gowrap1()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:1851 +0x44
  
  Goroutine 471 (running) created at:
    github.com/voedger/voedger/pkg/isequencer.(*sequencer).startActualizer()
        /home/runner/work/voedger/voedger/pkg/isequencer/impl.go:83 +0x1ad
    github.com/voedger/voedger/pkg/isequencer.(*sequencer).Actualize()
        /home/runner/work/voedger/voedger/pkg/isequencer/impl.go:526 +0x3e4
    github.com/voedger/voedger/pkg/isequencer_test.TestISequencer_LongRecovery()
        /home/runner/work/voedger/voedger/pkg/isequencer/isequencer_test.go:938 +0x4c9
    testing.tRunner()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:1792 +0x225
    testing.(*T).Run.gowrap1()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:1851 +0x44
  
  Goroutine 288 (running) created at:
    testing.(*T).Run()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:1851 +0x8f2
    testing.runTests.func1()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:2279 +0x85
    testing.tRunner()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:1792 +0x225
    testing.runTests()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:2277 +0x96c
    testing.(*M).Run()
        /opt/hostedtoolcache/go/1.24.2/x64/src/testing/testing.go:2[142](https://github.com/voedger/voedger/actions/runs/14372813004/job/40298919008#step:7:144) +0xeea
    main.main()
        _testmain.go:85 +0x164
  ==================
  --- FAIL: TestISequencer_LongRecovery (0.07s)
      testing.go:1490: race detected during execution of test
  FAIL
```

--- a/pkg/isequencer/impl.go
+++ b/pkg/isequencer/impl.go
@@ -10,7 +10,6 @@ import (
 	"errors"
 	"fmt"
 	"maps"
-	"sync"
 
 	"github.com/voedger/voedger/pkg/coreutils"
 )
@@ -70,7 +69,6 @@ func (s *sequencer) startFlusher() {
 
 	flusherCtx, flusherCtxCancel := context.WithCancel(context.Background())
 	s.flusherCtxCancel = flusherCtxCancel
-	s.flusherWG = &sync.WaitGroup{}
 	s.flusherWG.Add(1)
 	go s.flusher(flusherCtx)
 }
@@ -84,11 +82,8 @@ func (s *sequencer) startActualizer() {
 }
 
 func (s *sequencer) stopFlusher() {
-	if s.flusherWG != nil {
-		s.flusherCtxCancel()
-		s.flusherWG.Wait()
-		s.flusherWG = nil
-	}
+	s.flusherCtxCancel()
+	s.flusherWG.Wait()
 }
 
 // signalToFlushing is used to signal the flusher to start flushing.
@@ -129,11 +124,6 @@ func (s *sequencer) flusher(ctx context.Context) {
 		s.flusherWG.Done()
 	}()
 
-	// Non-blocking write to flusherStartedCh for tests
-	select {
-	case s.flusherStartedCh <- struct{}{}:
-	default:
-	}
 	// Wait for ctx.Done()
 	for ctx.Err() == nil {
 		select {
@@ -379,10 +369,7 @@ func (s *sequencer) actualizer(actualizerCtx context.Context) {
 		s.actualizerInProgress.Store(false)
 	}()
 
-	// if s.flusherWG is not nil
-	if s.flusherWG != nil {
-		s.stopFlusher()
-	}
+	s.stopFlusher()
 
 	// Clean s.toBeFlushed, toBeFlushedOffset
 	s.toBeFlushedMu.Lock()
@@ -534,10 +521,7 @@ func (s *sequencer) cleanup() {
 		s.actualizerInProgress.Store(false)
 	}
 
-	if s.flusherWG != nil {
-		s.flusherCtxCancel()
-		s.flusherWG.Wait()
-	}
+	s.stopFlusher()
 
 	s.cleanupCtxCancel()
 }

--- a/pkg/isequencer/provide.go
+++ b/pkg/isequencer/provide.go
@@ -47,7 +47,7 @@ func New(params Params, seqStorage ISeqStorage, iTime coreutils.ITime) (ISequenc
 		cleanupCtx:       cleanupCtx,
 		cleanupCtxCancel: cleanupCtxCancel,
 		iTime:            iTime,
-		flusherStartedCh: make(chan struct{}, 1),
+		flusherCtxCancel: func() {}, // flusher is not started -> prevent nil
 		flusherSig:       make(chan struct{}, 1),
 		actualizerWG:     &sync.WaitGroup{},
 		seqStorage:       seqStorage,

--- a/pkg/isequencer/types.go
+++ b/pkg/isequencer/types.go
@@ -77,13 +77,10 @@ type sequencer struct {
 	// Closes when flusher needs to be stopped
 	flusherCtxCancel context.CancelFunc
 	// Used to wait for flusher goroutine to exit
-	// Set to nil when flusher is not running
 	// Is not accessed concurrently since
 	// - Is accessed by actualizer() and cleanup()
 	// - cleanup() first shutdowns the actualizer() then flusher()	flusherWg sync.WaitGroup
-	flusherWG *sync.WaitGroup
-	// Used in tests to signal that flusher is started
-	flusherStartedCh chan struct{}
+	flusherWG sync.WaitGroup
 	// Buffered channel [1] to signal flusher to flush
 	// Written (non-blocking) by Flush()
 	flusherSig chan struct{}

